src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,252 @@
+}
+
+//Mimic MUI header style for footer
+const TableCellStyleOverrides: TableCellProps["sx"] = {
+    fontFamily: "var(--defaultFont)",
+    backgroundColor: "var(--secondaryBackground)",
+    fontWeight: "bold",
+    color: "var(--secondaryForeground)",
+};
+
+const calculateFooterRow = (cols: Column[], rows: RowData[]): FooterRow => {
+    const totalCols = cols.filter((x) => x.columnCalculation);
+    const total: FooterRow = {};
+    for (const col of totalCols) {
+        if (col.columnCalculation) {
+            const values = rows.map((x) => x[col.name]);
+            if (col.columnCalculation === "sum") {
+                total[col.name] = calculateSum(values);
+            } else if (col.columnCalculation === "average") {
+                total[col.name] = calculateAverage(values);
+            } else {
+                total[col.name] = col.columnCalculation(values);
+            }
+        }
+    }
+    return total;
+};
+
+const calculateSum = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    return numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+};
+
+const calculateAverage = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    const sum = numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+    return sum / numericVals.length || 0;
+};
+/**
+ * A table form element.
+ * @displayName Editable Table
+ * @description A table form element that allows for inline editing and tabular calculation.
+ * @param props The props that will be provided by the Workflow runtime.
+ */
+function EditableTableElement(props: EditableTableElementProps): React.ReactElement | null {
+    const {
+        rows,
+        cols,
+        maxHeight,
+        maxWidth,
+        onMouseEnter,
+        onMouseLeave,
+        raiseEvent,
+        setValue,
+        size,
+        stickyHeader,
+        footerLabel,
+    } = props;
+    const [data, setData] = useState(rows);
+    setValue(data);
+    useEffect(() => {
+        setData(rows);
+    }, [rows]);
+
+    const illegalCols = cols.filter(
+        (x) => !!x.type && ["text", "number", "date", "time", undefined].indexOf(x.type) === -1
+    );
+    if (illegalCols.length > 0) {
+        for (const illegalCol of illegalCols) {
+            const type = illegalCol.type as string;
+            console.log(
+                `Unsupported column type ${type} in column ${illegalCol.name}.  Supported types: text, number.`
+            );
+            illegalCol.type = undefined;
+        }
+    }
+    const includeFooter = cols.some((x) => x.columnCalculation);
+    const footerRow: FooterRow = calculateFooterRow(cols, rows);
+
+    const handleMouseEnter = (event: React.MouseEvent, row: RowData) => {
+        raiseEvent("custom", {
+            eventType: "mouseEnter",
+            rowData: { rows: data, row: row },
```

Comment:
Is there a reason not to have that same signature as the `Table` element? What do we need `rows` for?